ts/services/ElementService.ts

```diff
@@ -179,6 +157,118 @@ class ElementService {
     return response;
   }
 
+  public async elementSearch(config, logger): Promise<void> {
+
+    // base query
+    const requestBody = this.buildBaseElementQuery(config);
+
+    // add name term
+    if (typeof config.name !== 'undefined') {
+      const field = 'elementNames';
+      requestBody.body[field] = this.buildValueQueryTerm(config.name, true, false);
+    }
+    // add type term
+    if (typeof config.type !== 'undefined') {
+      const field = 'elementTypes';
+      requestBody.body[field] = this.buildValueQueryTerm(config.type, true, true);
+    }
+    // add attributes term
+    if (typeof config.attribute !== 'undefined') {
+      const attrKey = config.attribute.split('=')[0];
+      const attrVal = config.attribute.split('=')[1];
+
+      const field = 'attributes';
+      requestBody.body[field] = this.buildKeyValueQueryTerm(attrKey, attrVal);
+    }
+    // add elementTags term
+    if (typeof config.tag !== 'undefined') {
+      const tagKey = config.tag.split('=')[0];
+      const tagVal = config.tag.split('=')[1];
+
+      const field = 'elementTags';
+      requestBody.body[field] = this.buildKeyValueQueryTerm(tagKey, tagVal);
+    }
+    // add collectors term
+    if (typeof config.collector !== 'undefined') {
+      const field = 'collectorNames';
+      requestBody.body[field] = this.buildValueQueryTerm(config.collector, true, true);
+    }
+
+    logger.debug(JSON.stringify(requestBody, null, 2));
+
+    try {
+      const response = await request(requestBody);
+      if (config.format === 'text') {
+        logger.info(`###########################################################################`);
+        logger.info('## Element Search');
+        logger.info(`##  Page Size [${response.page.size}]`);
+        logger.info(`##  Page Number [${response.page.number} of ${response.page.totalPages - 1}]`);
+        logger.info(`##  Total Elements [${response.page.totalElements}]`);
+        logger.info(`###########################################################################`);
+        logger.info(response.page.content.map((el) => {
+          return `name[${el.name}] id[${el.id}] fqn[${el.fqn}] type[${el.type}]`;
+        }));
+      }
+      if (config.format === 'json') {
+        logger.info(JSON.stringify(response, null, 2));
+      }
+    } catch (e) {
+      logger.error('There was an error listing the elements: ' + e);
+    }
+  }
+
+  public buildBaseElementQuery(config) {
+    const queryPage = config.page ? config.page : 0;
+    const queryPageSize = config.pageSize ? config.pageSize : 35;
+
+    return {
+      auth: {
+        pass: config.password,
+        user: config.username
+      },
+      body: {
+        endDate: moment().format(),
+        page: queryPage,
+        pageSize: queryPageSize,
+        sort: {
+          field: 'name',
+          missing: '_last',
```

Comment:
I don't think this does anything in these basic queries and can be removed.

Reply:
I confirmed that this sorts the results by name, or do you mean just the `missing` part of the query?

Reply:
Yes, sorry. Only the `missing` line.

Reply:
okay, removed...